### DISCUSSION TITLE
utoipa-3.0.0

### DISCUSSION THREAD

**Discussion by juhaku** [juhaku/utoipa#462] [Announcements]
## What's New :gem: :new: :tada:

About a 1 year ago the first public release of utoipa was published. A 1 day less to be exact. It has been a long journey since the first release and the utoipa has seen countless improvements, even coming from the latest release a numerous things have been tinkered.

The API has seen quite a reform and it is unified across the board in many places. Missing attributes, configuration options and functionalities have been added to many places. Docs were improved a lot, a few bug fixes here and there and 2 new derive implementations were added.

Huge thanks to you dear contributor for your support and help in this project as well as those who have filed issues and created discussions on important matters that help to make this library better. :1st_place_medal: 

This is a breaking release and there are few things to read through when you update to this release. You can read more from here: https://github.com/juhaku/utoipa/discussions/456

* (d19aadf) Refactor `ToReponse` trait (#460) @juhaku 
* (11288c3) Refactor to schema casting as (#459) @juhaku 
* (5b51eb4) Enhance generic aliases with lifetimes support (#458) @juhaku 
* (46fe673) Enhance path tuple argument support (#455) @juhaku 
* (2e13074) Fix spelling (#450) @jayvdb 
* (b167838) Add support for serde `skip_serializing` (#438) @Xenfo 
* (13a3aae) Update versions
* (50ab67f) Update README
* (f37261e) Update docs
* (5a78fef) Improve docs (#444) @juhaku 
* (084b2a1) Enhance reponses derive support (#443) @juhaku 
* (284b01a) Document `preserve_order` feature flag (#437) @SanchithHegde 
* (2a5c09d) Add `preserve_order` feature to preserve property order during serialization (#436) @SanchithHegde 
* (4a08316) Add derive `IntoResponses` support (#433) @juhaku 
* (28e64ad) Feat/serde enum representation (#414) @AlisCode 
* (e50da56) Fix empty string path parameter name for Axum (#424) @jay3332 
* (571fc10) Enhance `ToResponse` implementation (#419) @juhaku 
* (d8c5ec5) Swagger UI url config (#418) @juhaku 
* (b0f2eb1) Fix typo in doc
* (3d0ea69) Add `ToResponse` derive implementation (#416) @juhaku 
* (ddb9eb3) Addresss clippy lint
* (6c7f6a2) Improve documenetation
* (5b643af) Enhance repeated attributes support (#411) @juhaku 
* (1af2443) Add support for repeated `schema(...)` definition (#410) @juhaku 
* (71f46ec) Add external ref(...) attribute (#409) @juhaku 
* (7138fd9) Make derive OpenApi server variable names LitStr
* (79ab858) Refactor `Type` to `TypeTree` (#408) @juhaku 
* (db19971) Fix make untagged enum object variants required (#407) @juhaku 
* (abf4728) Add example attributes for request body (#406) @juhaku 
* (d4489d1) Add auto detect application/octet-stream type (#405) @juhaku 
* (40b8bc0) Add support for chrono NaiveDate (#404) @juhaku 
* (8b541cf) Add support for multiple examples in response (#403) @juhaku 
* (8e3bed2) Add Example type to OpenApi types (#402) @juhaku 
* (9639a06) Fix time-crate typo in schema format tokens (#401) @juhaku 
* (30a45d8) Add derive info support for derive OpenApi (#400) @juhaku 
* (4d982c6) Add `merge` functionality for `OpenApi` (#397) @juhaku 
* (7026f9e) Add derive servers attribute for OpenApi (#395) @juhaku 
* (9a482c9) Fix primitive type generic aliases (#393) @juhaku 
* (a4b1af0) Add support for unit sructs (#392) @juhaku 
* (bb1717b) Add support for `schema_with` custon fn reference (#390) @juhaku 
* (391daef) Add support for multiple serde definitions (#389) @juhaku 
* (0cf8eae) Add support for tuple Path parameters for axum (#388) @juhaku 
* (fdd244c) Fix `TypeTree` for `slice` and `array` types (#387) @juhaku 
* (7661aab) Add derive validation attributes to `IntoParams` (#386) @juhaku 
* (fed7237) Update `ToSchema` documentation
* (23f517c) Add support for derive validation attributes (#385) @juhaku 
* (2679108) feat: make schema_type and title pub on Object (#382) @kyle-mccarthy 
* (daa9f43) Use BTreeMap for responses of components to make it fixed order (#380) @RossyWhite 
* (ae840e3) Add support for missing attributes for validation (#379)@juhaku 
* (772c089) Chore make `serde_json` mandatory dependency (#378) @juhaku 
* (093014e) Add support for multiple return types (#377) @juhaku 
* (1df2773) Feature http status codes (#376) @juhaku 
* (63613c2) Add support for self referencing schema (#375) @juhaku 
* (483b778) Add missing features to `IntoParams` (#374) @juhaku 
* (436cccb) Refactor some path derive with `IntoParmas` tests
* (afa62d2) Update utoipa-swagger-ui install example
* (65842b9) Chore refine description attribute (#373) @juhaku 
* (67b9585) Fix minum axum version info
* (6971fb3) Update swagger-ui dependencies versions
* (5b2c8ef) Update utoipa-swagger-ui version
* (badb475) cargo format
* (a6f41cf) Fix broken swagger-ui on axum  (#370) @simmsb 
* (b558b36) Update to axum 0.6.0 (#369) @davidpdrsn 

#### Full [change log](https://github.com/juhaku/utoipa/compare/utoipa-2.4.2...utoipa-3.0.0)

<hr /><em>This discussion was created from the release <a href='https://github.com/juhaku/utoipa/releases/tag/utoipa-3.0.0'>utoipa-3.0.0</a>.</em>

**Comment by jayvdb** [comment]
https://github.com/juhaku/utoipa/pull/424 was [jay3332](https://github.com/jay3332), not me.

**Reply by juhaku** [reply]
Oh no, did I make a mistake. Thanks for correction
